core/src/main/scala/kafka/server/NodeToControllerChannelManager.scala

```diff
@@ -262,22 +227,6 @@ class NodeToControllerRequestThread(
 
   this.logIdent = logPrefix
 
-  private def maybeResetNetworkClient(controllerInformation: ControllerInformation): Unit = {
-    if (isNetworkClientForZkController != controllerInformation.isZkController) {
-      debug("Controller changed to " + (if (isNetworkClientForZkController) "kraft" else "zk") + " mode. " +
-        s"Resetting network client with new controller information : ${controllerInformation}")
-      // Close existing network client.
-      val oldClient = networkClient
-      oldClient.initiateClose()
-      oldClient.close()
-
-      isNetworkClientForZkController = controllerInformation.isZkController
-      updateControllerAddress(controllerInformation.node.orNull)
-      controllerInformation.node.foreach(n => metadataUpdater.setNodes(Seq(n).asJava))
-      networkClient = networkClientFactory(controllerInformation)
```

Comment:
we don't change `networkClient` anymore, so `networkClient` can be declared as final variable.